xgi/algorithms/simpliciality.py

```diff
@@ -0,0 +1,434 @@
+            s = _count_subfaces(t, e, min_size=min_size)
+            m = _max_number_of_subfaces(min_size, len(e))
+            d = m - s  # missing subfaces
+            if normalize and m != 0:
+                d *= 1.0 / m
+            avg_d += d / len(max_faces)
+    return avg_d
+
+
+def simplicial_fraction(H, min_size=2, exclude_min_size=True):
+    """Computing the simplicial fraction for a hypergraph.
+
+    What fraction of the hyperedges are simplices?
+
+    Parameters
+    ----------
+    H : Hypergraph
+        The hypergraph of interest
+    min_size : int, optional
+        The minimum edge size to consider a simplex, by default 2
+    exclude_min_size : bool, optional
+        Whether to include minimal simplices when counting simplices, by default True
+
+    Returns
+    -------
+    float
+        The simplicial fraction
+
+    References
+    ----------
+    "The simpliciality of higher-order order networks"
+    by Nicholas Landry, Jean-Gabriel Young, and Nicole Eikmeier,
+    *EPJ Data Science* **13**, 17 (2024).
+    """
+    try:
+        ns = _count_simplices(H, min_size, exclude_min_size)
+        ps = _potential_simplices(H, min_size, exclude_min_size)
+        return ns / ps
+    except ZeroDivisionError:
+        return np.nan
+
+
+#### Helper functions
+def _powerset(iterable, min_size=1, max_size=None):
+    """Generates a modified powerset.
+
+    User can specify the maximum and minimum size
+    of the sets in the powerset.
+
+    Parameters
+    ----------
+    iterable : iterable
+        The set for which to compute the powerset.
+    min_size: int, default: 1
+        The minimum hyperedge size to include when
+        calculating whether a hyperedge is a simplex
+        by counting subfaces.
+    max_size : int, default: None.
+        The maximum size to include when computing
+        the power set. When max_size=None, it generates
+        the powerset including the edge itself.
+
+    Returns
+    -------
+    itertools.chain
+        a generator of the sets in the powerset.
+    """
+    s = iterable
+    if max_size is None:
+        max_size = len(s)
+
+    return chain.from_iterable(
+        combinations(s, r) for r in range(min_size, max_size + 1)
+    )
+
+
+def _count_subfaces(t, face, min_size=1):
+    """Computing the edit distance for a single face.
+
+    Parameters
+    ----------
+    t : Trie
+        The trie representing the hypergraph
+    face : iterable
+        The edge for which to find the edit distance
+    min_size: int, default: 1
+        The minimum hyperedge size to include when
+        calculating whether a hyperedge is a simplex
+        by counting subfaces.
+
+    Returns
+    -------
+    int
+        The edit distance
+    """
+    sub_edges = list(_powerset(face, min_size=min_size, max_size=len(face) - 1))
+    count = 0
+    for e in sub_edges:
+        if t.search(e):
+            count += 1
+
+    return count
+
+
+def _count_subfaces(t, face, min_size=1):
```

Comment:
Is this a repeat of the above function with the same signature?

Reply:
Oops --- great catch. Fixed now!